src/hotspot/share/runtime/os.hpp

```diff
@@ -686,7 +686,11 @@ class os: AllStatic {
   // return current frame. pc() and sp() are set to NULL on failure.
   static frame      current_frame();
 
-  static void print_hex_dump(outputStream* st, address start, address end, int unitsize);
+  static void print_hex_dump(outputStream* st, address start, address end, int unitsize,
+                             int bytes_per_line, address logical_start);
+  static void print_hex_dump(outputStream* st, address start, address end, int unitsize) {
+    print_hex_dump(st, start, end, unitsize, /*bytes_per_line=*/16, /*logical_start=*/start);
+  }
```

Comment:
This is a bit unclear. Could we do it like this instead:

// Print a hex dump from [start .. end). Output lines are prefixed with addresses.
static void print_hex_dump(outputStream* st, address start, address end, int unitsize)


// Print a hex dump from [base+start .. base+end). Output lines are prefixed with offset in relation to base.
static void print_hex_dump(outputStream* st, address base, intx start, intx end, int unitsize)

Reply:
But this is not what I want. I am calling with parameters like:

```
print_hex_dump(st, /*start=*/0x7ffff0000, end, unitsize, bytes_per_line,  /*logical_start=*/0x800000000);
```
So my buffer is at some random address, but I want the print out to pretend that it started at 0x800000000.

Reply:
Okay. A bit special, usually one would base it at 0 (e.g. printing network packets or similar). I think its okay then. We can revisit this later if needed.